### PR TITLE
dhcp: introduce json_dispatch_client_id() and friends

### PR DESCRIPTION
**Pull request by yuwata**
split-out of #30021.

### CODE REVIEW

File: src/shared/json.c [hunk 1]
```diff
@@ -4993,6 +4995,60 @@ int json_dispatch_unbase64_iovec(const char *name, JsonVariant *variant, JsonDis
...
+        size_t sz, k = 0;
+
+        assert(variant);
+
+        if (!json_variant_is_array(variant))
+                return json_log(variant, flags, SYNTHETIC_ERRNO(EINVAL), "JSON field '%s' is not an array.", strna(name));
+
+        sz = json_variant_elements(variant);
+
+        buffer = new(uint8_t, sz);
```
Comment (poettering) [inline src/shared/json.c:5010]: doesn't matter, but for other cases like this we pretty systematically append an extra NUL byte just for safety reasons to the end of decoded buffers. see unbase64mem() for example. i.e. maybe just replace sz by sz+1 here, and end the buffer in a NUL byte, simply for paranoia.

Comment (yuwata) [reply src/shared/json.c:5010]: Yeah, I considered that, but this is for binary data... But OK, I will add one more byte.

Comment (poettering) [reply src/shared/json.c:5010]: yeah, same as unbase64mem() actually. but like unbase64mem() this is super generic code, and for thta kind of stuff i'd always try to be defensive.

but doesn't matter

Comment (yuwata) [reply src/shared/json.c:5010]: -> #30893.

**Pull request merged**